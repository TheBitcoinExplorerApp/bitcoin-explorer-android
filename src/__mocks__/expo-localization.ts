const getLocales = () => [
  { country: "pt", languageTag: "pt-BR", languageCode: "pt", isRTL: false },
];

export default getLocales;
