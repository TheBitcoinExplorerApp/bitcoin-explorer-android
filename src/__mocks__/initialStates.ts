import { BlockType } from "src/components/BlocksInfo/components/Block/type";
import { AddressInfoType, TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks: BlockType[] = [
  {
    blockHash: "",
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,

    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};

export const initialStateFormattedTransactions = [
  {
    transactionId: "0",
    value: 0,
    fee: 0,
  },
  {
    transactionId: "1",
    value: 0,
    fee: 0,
  },
];

export const initialStateAddress: AddressInfoType = {
  address: "",
  balance: 0,
  totalAmountReceived: 0,
  totalAmountSent: 0,
};
