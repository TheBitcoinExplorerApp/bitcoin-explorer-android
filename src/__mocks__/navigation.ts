const mockNavigation = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNavigation = jest.requireActual("@react-navigation/native");
  return {
    ...actualNavigation,
    useNavigation: () => ({
      navigate: mockNavigation,
    }),
  };
});
