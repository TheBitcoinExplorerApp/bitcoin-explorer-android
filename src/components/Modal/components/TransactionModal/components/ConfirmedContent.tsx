// eslint-disable-next-line import/no-extraneous-dependencies
import { View, StyleSheet } from "react-native";
import BoxContainerWithText from "src/components/BoxContainerWithText/BoxContainerWithText";
import { TransactionType } from "src/components/Modal/types";
import useAppDataStore from "src/context/DataProvider";
import { formatDate } from "src/utils/formatBlockInfo";

type ConfirmedContentProps = Required<
  Pick<TransactionType, "fee" | "size" | "statusTransaction">
>;

const styles = StyleSheet.create({
  container: {
    gap: 16,
  },
});

export default function ConfirmedContent(props: ConfirmedContentProps) {
  const { fee, size, statusTransaction } = props;
  const { blockHeight, blockTime } = statusTransaction;
  const { i18n } = useAppDataStore();

  const { day, month, year, hour, minutes } = formatDate(blockTime);

  return (
    <View style={styles.container}>
      <View
        style={{
          flexDirection: "row",
          justifyContent: "space-between",
        }}
      >
        <BoxContainerWithText
          firstText={`${i18n.t("block")} ${blockHeight}`}
          secondText=""
          width="auto"
        />

        <BoxContainerWithText
          firstText={i18n.t("confirmed")}
          secondText=""
          width="auto"
        />
      </View>

      <View>
        <BoxContainerWithText
          firstText={i18n.t("date_time")}
          secondText={`${day}/${month}/${year} ${hour}:${minutes}`}
          borderStyles={{
            borderBottomEndRadius: 0,
            borderBottomStartRadius: 0,
            borderBottomColor: "#D9D9D9",
          }}
        />
        <BoxContainerWithText
          firstText={i18n.t("size")}
          secondText={`${size} B`}
          borderStyles={{
            borderBottomEndRadius: 0,
            borderBottomStartRadius: 0,
            borderTopEndRadius: 0,
            borderTopStartRadius: 0,
            borderBottomColor: "#D9D9D9",
          }}
        />
        <BoxContainerWithText
          firstText={i18n.t("fee")}
          secondText={`${(fee / 100000000).toString()} BTC`}
          borderStyles={{
            borderTopEndRadius: 0,
            borderTopStartRadius: 0,
          }}
        />
      </View>
    </View>
  );
}
