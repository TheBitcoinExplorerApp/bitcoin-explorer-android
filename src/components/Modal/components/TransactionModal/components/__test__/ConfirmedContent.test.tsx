/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import { initialStateTransaction } from "src/__mocks__/initialStates";
import ConfirmedContent from "../ConfirmedContent";

describe("ConfirmedContent", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ConfirmedContent
          fee={initialStateTransaction.fee}
          size={initialStateTransaction.size}
          statusTransaction={{
            confirmed: true,
            blockHeight: 1,
            blockHash: "abc",
            blockTime: 204,
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
