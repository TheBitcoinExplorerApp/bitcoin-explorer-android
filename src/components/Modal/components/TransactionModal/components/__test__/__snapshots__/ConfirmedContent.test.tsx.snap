// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmedContent renders correctly 1`] = `
<View
  style={
    {
      "gap": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "height": 48,
            "justifyContent": "space-between",
            "paddingHorizontal": 18,
            "paddingVertical": 9,
          },
          {
            "borderBottomColor": undefined,
            "borderBottomEndRadius": undefined,
            "borderBottomStartRadius": undefined,
            "borderBottomWidth": 0,
            "borderRadius": 7,
            "borderTopEndRadius": undefined,
            "borderTopStartRadius": undefined,
            "flexDirection": "row",
            "width": "auto",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        Bloco 1
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
            "fontSize": 15,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "height": 48,
            "justifyContent": "space-between",
            "paddingHorizontal": 18,
            "paddingVertical": 9,
          },
          {
            "borderBottomColor": undefined,
            "borderBottomEndRadius": undefined,
            "borderBottomStartRadius": undefined,
            "borderBottomWidth": 0,
            "borderRadius": 7,
            "borderTopEndRadius": undefined,
            "borderTopStartRadius": undefined,
            "flexDirection": "row",
            "width": "auto",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        Confirmada
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
            "fontSize": 15,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "height": 48,
            "justifyContent": "space-between",
            "paddingHorizontal": 18,
            "paddingVertical": 9,
          },
          {
            "borderBottomColor": "#D9D9D9",
            "borderBottomEndRadius": 0,
            "borderBottomStartRadius": 0,
            "borderBottomWidth": 0.1,
            "borderRadius": 7,
            "borderTopEndRadius": undefined,
            "borderTopStartRadius": undefined,
            "flexDirection": "row",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        Data/Hora
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
            "fontSize": 15,
          }
        }
      >
        31/12/1969 21:3
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "height": 48,
            "justifyContent": "space-between",
            "paddingHorizontal": 18,
            "paddingVertical": 9,
          },
          {
            "borderBottomColor": "#D9D9D9",
            "borderBottomEndRadius": 0,
            "borderBottomStartRadius": 0,
            "borderBottomWidth": 0.1,
            "borderRadius": 7,
            "borderTopEndRadius": 0,
            "borderTopStartRadius": 0,
            "flexDirection": "row",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        Tamanho
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
            "fontSize": 15,
          }
        }
      >
        0 B
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "height": 48,
            "justifyContent": "space-between",
            "paddingHorizontal": 18,
            "paddingVertical": 9,
          },
          {
            "borderBottomColor": undefined,
            "borderBottomEndRadius": undefined,
            "borderBottomStartRadius": undefined,
            "borderBottomWidth": 0,
            "borderRadius": 7,
            "borderTopEndRadius": 0,
            "borderTopStartRadius": 0,
            "flexDirection": "row",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 15,
          }
        }
      >
        Taxa
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
            "fontSize": 15,
          }
        }
      >
        0 BTC
      </Text>
    </View>
  </View>
</View>
`;
